tests/PhpDocumentor/ReflectionExtractorTest.php

```diff
@@ -78,4 +84,63 @@ public function test_it_throws_missing_service_id_exception_if_there_is_no_servi
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'foo', '\\Foo', false), $serviceProperties[0]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'bar', '\\Bar', false), $serviceProperties[1]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'baz', '\\Baz', true), $serviceProperties[2]);
+    }
+
+    public function test_it_extracts_service_definitions_from_overriden_public_properties()
+    {
+        $serviceProperties = $this->servicePropertyExtractor->extract(OverridePublicPropertyExample::class);
+
+        $this->assertCount(5, $serviceProperties);
+        $this->assertEquals(new Property(OverridePublicPropertyExample::class, 'foo', 'bar', false), $serviceProperties[0]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'bar', '\\Bar', false), $serviceProperties[1]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'foo', '\\Foo', false), $serviceProperties[2]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'bar', '\\Bar', false), $serviceProperties[3]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'baz', '\\Baz', true), $serviceProperties[4]);
+    }
+
+    public function test_it_extracts_service_definitions_from_overriden_protected_properties()
+    {
+        $serviceProperties = $this->servicePropertyExtractor->extract(OverrideProtectedPropertyExample::class);
+
+        $this->assertCount(5, $serviceProperties);
+        $this->assertEquals(new Property(OverrideProtectedPropertyExample::class, 'bar', 'something_else', false), $serviceProperties[0]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'foo', '\\Foo', false), $serviceProperties[1]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'foo', '\\Foo', false), $serviceProperties[2]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'bar', '\\Bar', false), $serviceProperties[3]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'baz', '\\Baz', true), $serviceProperties[4]);
+    }
+
+    public function test_it_extracts_service_definitions_from_overriden_private_properties()
+    {
+        $serviceProperties = $this->servicePropertyExtractor->extract(OverridePrivatePropertyExample::class);
+
+        $this->assertCount(6, $serviceProperties);
+        $this->assertEquals(new Property(OverridePrivatePropertyExample::class, 'baz', 'something_else', true), $serviceProperties[0]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'foo', '\\Foo', false), $serviceProperties[1]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'bar', '\\Bar', false), $serviceProperties[2]);
+        $this->assertEquals(new Property(PropertyVisibilityExample::class, 'foo', '\\Foo', false), $serviceProperties[3]);
```

Comment:
Not sure if duplicated properties should be filtered out here or in the injector (currently done in the injector with `unset...`)